diff --git a/libdispatch/dwinpath.c b/libdispatch/dwinpath.c
index b83ef543ce..18681a9063 100644
--- a/libdispatch/dwinpath.c
+++ b/libdispatch/dwinpath.c
@@ -8,6 +8,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <assert.h>
+#include <errno.h>
 #ifdef HAVE_FCNTL_H
 #include <fcntl.h>
 #endif
@@ -155,7 +156,7 @@ static char*
 makeabsolute(const char* relpath)
 {
     char* path = NULL;
-#ifdef _MSC_VER
+#if defined(_MSC_VER) || defined(_WIN32)
     path = _fullpath(NULL,relpath,8192);
 #else
     path = realpath(relpath, NULL);
@@ -214,7 +215,7 @@ NCaccess(const char* path, int mode)
     int status = 0;
     char* cvtname = NCpathcvt(path);
     if(cvtname == NULL) return -1;
-#ifdef _MSC_VER
+#if defined(_MSC_VER) || defined(_WIN32)
     status = _access(cvtname,mode);
 #else
     status = access(cvtname,mode);
diff --git a/libsrc/memio.c b/libsrc/memio.c
index aac8a3695c..3aa2190c72 100644
--- a/libsrc/memio.c
+++ b/libsrc/memio.c
@@ -17,7 +17,7 @@
 #ifdef HAVE_FCNTL_H
 #include <fcntl.h>
 #endif
-#ifdef _MSC_VER /* Microsoft Compilers */
+#if defined(_MSC_VER) || defined(_WIN32) /* Windows Compilers */
 #include <windows.h>
 #include <winbase.h>
 #include <io.h>
@@ -711,7 +711,7 @@ readfile(const char* path, NC_memio* memio)
     char* p = NULL;
 
     /* Open the file for reading */
-#ifdef _MSC_VER
+#if defined(_MSC_VER) || defined(_WIN32)
     f = NCfopen(path,"rb");
 #else
     f = NCfopen(path,"r");
@@ -761,7 +761,7 @@ writefile(const char* path, NCMEMIO* memio)
     char* p = NULL;
 
     /* Open/create the file for writing*/
-#ifdef _MSC_VER
+#if defined(_MSC_VER) || defined(_WIN32)
     f = NCfopen(path,"wb");
 #else
     f = NCfopen(path,"w");
